doc/concepts/morphology.rst

```diff
@@ -109,6 +125,7 @@ Segment trees
 .. glossary::
 
   segment tree
+  segment_tree
```

Comment:
Why a second segment_tree?

Reply:
I resolved the `m` prefix situation by adding a second term in every definition that has a typname that deviates. In the case of the segment tree, there is no `m` prefix, but the space is an `_`.